projects/storefrontapp-e2e-cypress/cypress/helpers/anonymous-consents.ts

```diff
@@ -0,0 +1,254 @@
+    cy.selectUserMenuOption({
+      option: 'Consent Management',
+    });
+
+    giveConsent();
+    signOutUser();
+    checkBanner();
+
+    clickViewDetailsFromBanner();
+    giveOrRemoveAnonymousConsent(2);
+    closeDialog();
+
+    const loginPage = waitForPage('/login', 'getLoginPage');
+    cy.getByText(/Sign in \/ Register/i).click();
+    cy.wait(`@${loginPage}`);
+
+    login(
+      standardUser.registrationData.email,
+      standardUser.registrationData.password
+    );
+    checkBanner();
+
+    cy.selectUserMenuOption({
+      option: 'Consent Management',
+    });
+
+    cy.get('input[type="checkbox"]')
+      .first()
+      .should('be.checked');
+  });
+}
+
+export function movingFromAnonymousToRegisteredUser() {
+  it('should transfer anonoymous consents when registered', () => {
+    openDialogUsingFooterLink();
+    giveOrRemoveAnonymousConsent(2);
+    closeDialog();
+
+    registerUserAndCheckMyAccountConsent(false, 2);
+  });
+}
+
+// export function movingFromLoggedInUserToAnonymousUser() {
+//   it('should restore anonoymous consents when logging out', () => {
+//     const newUser = registerUser();
+//     cy.visit(PAGE_URL_LOGIN);
+//     login(newUser.email, newUser.password);
+//     signOutUser();
+
+//     openDialogUsingFooterLink();
+//     giveOrRemoveAnonymousConsent(2);
+//     closeDialog();
+
+//     cy.visit(PAGE_URL_LOGIN);
+//     login(newUser.email, newUser.password);
+//     signOutUser();
+
+//     openDialogUsingFooterLink();
+//     cy.get('.cx-toggle-text:nth-child(2)').should('contain.text', 'ON');
+//   });
+// }
+
+export function changeLanguageTest() {
+  it('should pull the new consent templates but preserve the consents state', () => {
+    openDialogUsingFooterLink();
+    acceptAllConsent();
+
+    siteContextChange('/', LANGUAGES, LANGUAGE_DE, LANGUAGE_LABEL);
+    openDialogUsingFooterLink();
+    checkConsentsGiven();
+  });
+}
+
+export function registerUserAndCheckMyAccountConsent(
+  consentCheckBox,
+  position
+) {
+  const newUser = registerUser(consentCheckBox);
+  login(newUser.email, newUser.password);
+  checkBanner();
+  cy.selectUserMenuOption({
+    option: 'Consent Management',
+  });
+  cy.get(`input[type="checkbox"]:nth-child(${position})`)
+    .first()
+    .should('be.checked');
+}
+
+export function giveRegistrationConsentTest() {
+  it('should give the registration consent', () => {
+    registerUserAndCheckMyAccountConsent(true, 1);
+  });
+}
+
+export function testAsAnonymousUser() {
+  it('should be able to see the banner', () => {
+    seeBannerAsAnonymous();
+  });
+
+  it('should close the banner and give all consents by clicking on "ALLOW ALL" in the banner', () => {
+    clickAllowAllFromBanner();
+    checkBannerHidden();
+  });
+
+  it('should click the footer to check if all consents were accepted and withdraw all consents afterwards', () => {
+    // cy.wait(5000);
```

Comment:
Temporary?

Reply:
yes was a test because something is broken